string intent
string place
string raw_text
